clients/dotnet/WebClient/MemoryWebClient.cs

```diff
@@ -438,11 +438,18 @@ private static (string contentType, long contentLength, DateTimeOffset lastModif
             // Add files to the form
             for (int i = 0; i < uploadRequest.Files.Count; i++)
             {
+                using Stream fileStream = uploadRequest.Files[i].FileContent;
```

Comment:
I would avoid introducing another IDisposable/IAsyncDisposable object to manage. 

What about this?
```csharp
if (uploadRequest.Files[i].FileContent is { CanSeek: true, Position: > 0 })
{
    uploadRequest.Files[i].FileContent.Seek(0, SeekOrigin.Begin);
}
```

Reply:
👍🏼